diffkemp/simpll/DifferentialFunctionComparator.cpp

```diff
@@ -1436,19 +1458,43 @@ int DifferentialFunctionComparator::cmpMemset(const CallInst *CL,
             return Res;
     }
 
+    // If the structure sizes are equal, we can end right away
     if (!cmpValues(CL->getArgOperand(2), CR->getArgOperand(2)))
         return 0;
 
-    // Get the struct types of memset destinations.
-    StructType *STyL = getStructType(CL->getOperand(0));
-    StructType *STyR = getStructType(CR->getOperand(0));
+    // Get the destination pointers
+    const Value *destL = CL->getArgOperand(0);
+    const Value *destR = CR->getArgOperand(0);
+
+    // If the destination is a bitcast, compare the original source value
+    const Value *ValL = isa<BitCastInst>(destL)
+                                ? dyn_cast<BitCastInst>(destL)->getOperand(0)
+                                : destL;
+    const Value *ValR = isa<BitCastInst>(destR)
+                                ? dyn_cast<BitCastInst>(destR)->getOperand(0)
+                                : destR;
+
+    // Retrieve the structure types
+#if LLVM_VERSION_MAJOR >= 15
+    DIType *STyL = getPointeeDITypeFromValue(ValL);
+    DIType *STyR = getPointeeDITypeFromValue(ValR);
+#else
+    StructType *STyL = getStructTypeFromValue(ValL);
+    StructType *STyR = getStructTypeFromValue(ValR);
+#endif
+
+    // Check if the structure types have the same names
+    if (!STyL || !STyR || STyL->getName() != STyR->getName())
+        return 1;
 
-    // Return 0 (equality) if both memory destinations are structs of the same
-    // name and each memset size is equal to the corresponding struct size.
-    return !STyL || !STyR
-           || cmpStructTypeSizeWithConstant(STyL, CL->getOperand(2))
-           || cmpStructTypeSizeWithConstant(STyR, CR->getOperand(2))
-           || STyL->getStructName() != STyR->getStructName();
+        // Check if the structure sizes match the memory size
+#if LLVM_VERSION_MAJOR >= 15
+    return cmpDITypeSizeWithConstant(STyL, CL->getOperand(2))
+           || cmpDITypeSizeWithConstant(STyR, CR->getOperand(2));
+#else
+    return cmpTypeSizeWithConstant(STyL, CL->getOperand(2))
+           || cmpTypeSizeWithConstant(STyR, CR->getOperand(2));
+#endif
```

Comment:
In fact, this code is very similar to the code above in `cmpAllocs`. Any chance it could be factored out?